cpp/src/interop/to_arrow.cu

```diff
@@ -378,13 +378,22 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<cudf::list_view>(
   auto children_meta =
     metadata.children_meta.empty() ? std::vector<column_metadata>{{}, {}} : metadata.children_meta;
   auto child_arrays = fetch_child_array(input_view, children_meta, ar_mr, stream);
-  if (child_arrays.empty()) {
+  if (child_arrays.empty() || child_arrays[0]->data()->length == 0) {
     // Empty list will have only one value in offset of 4 bytes
     auto tmp_offset_buffer = allocate_arrow_buffer(sizeof(int32_t), ar_mr);
     memset(tmp_offset_buffer->mutable_data(), 0, sizeof(int32_t));
 
+    std::shared_ptr<arrow::Array> values{nullptr};
+    std::shared_ptr<arrow::DataType> element_type{nullptr};
+    if (child_arrays.empty()) {
+      element_type = arrow::null();
+      values       = std::make_shared<arrow::NullArray>(0);
+    } else {
+      element_type = child_arrays[1]->type();
+      values       = child_arrays[1];
+    }
```

Comment:
```suggestion
    std::shared_ptr<arrow::Array> values = child_arrays.empty() ?  std::make_shared<arrow::NullArray>(0) : child_arrays[1];
    std::shared_ptr<arrow::DataType> element_type = child_arrays.empty() ? arrow::null() : std::make_shared<arrow::NullArray>(0);
```
Style nit. No need to initialize a variable just to reset it later.

Reply:
Made an additional simplification.